staging/src/k8s.io/client-go/tools/cache/shared_informer.go

```diff
@@ -305,7 +305,7 @@ func (s *sharedIndexInformer) AddEventHandlerWithResyncPeriod(handler ResourceEv
 			resyncPeriod = minimumResyncPeriod
 		}
 
-		if resyncPeriod < s.resyncCheckPeriod {
+		if resyncPeriod < s.resyncCheckPeriod || s.resyncCheckPeriod == 0 {
 			if s.started {
 				glog.Warningf("resyncPeriod %d is smaller than resyncCheckPeriod %d and the informer has already started. Changing it to %d", resyncPeriod, s.resyncCheckPeriod, s.resyncCheckPeriod)
```

Comment:
Just noticed: the text of this warning is technically incorrect if s.resyncCheckPeriod is 0, you're asking for a nonzero resyncPeriod, and s.started is true.

Reply:
Updated to prevent this log line from being emitted in this case... This will cause ```determineResyncPeriod()`` to emit "The specified resyncPeriod %v is invalid because this shared informer doesn't support resyncing" at shared_informer.go:276 which is a correct message in this case.